### PR TITLE
Feature/form components reworked

### PR DESCRIPTION
**Pull request by TheDadi**
Fixes # (issue).

Changes proposed in this pull request:
Introduce Form Components
 - a-checkbox
 - a-radio
 - a-choice
 - a-form-input
 - a-form-group

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
 
 # How Has This Been Tested?
 
 Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
 
 # Checklist:
 
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my own code
 - [x] I have commented my code, particularly in hard-to-understand areas
 - [x] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/m-form-group/index.scss [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  flex-shrink: 0;
+  flex-grow: 0;
+
+  cursor: pointer;
+}
+
+.m-form-group__icon {
+  display: block;
+  height: $form-group-icon-size;
```
Comment (AndyOGo) [inline src/components/m-form-group/index.scss]: `size` mixin could be used for equal witdhs and heigths

File: src/components/m-form-group/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+class AXAFormGroup extends BaseComponentGlobal {
+  static get observedAttributes() {
+    return ['label', 'info'];
+  }
+
+  constructor() {
+    super(styles, template);
+  }
+
+  connectedCallback() {
```
Comment (AndyOGo) [inline src/components/m-form-group/index.js]: @TheDadi 
For incremental rendering `connectedCallback` should be avoid until  really needed and replaced by `willRenderCallback` and `didRenderCallback`.
I refactored all components recently and documented stuff at https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTING.md#custom-elements

File: src/components/a-form-input/_template.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  placeholder = '',
+  value = '',
+  name,
+  disabled = false,
+}) {
+  return html`<div class="a-form-input__wrapper">
+      <input
+          id="${inputId}"
+          class="a-form-input__input" 
+          type="radio"
```
Comment (BarboraBardunova) [inline src/components/a-form-input/_template.js]: is it correct to have input with type="radio" + 2 lines below anything you pass as a parameter ? so when i use the webcomponent with type="email", it just transforms to type="radioemail"

File: src/components/m-form-group/index.scss [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  padding: $form-group-info-content-padding;
+
+  color: $color-prim-white;
+  background-color: $color-prim-blue-st-tropaz;
+
+  box-shadow: inset 0 0 15px $color-axa-blue;
+}
+
+.is-form-group-info-open {
```
[the remark this answers is not in the document]
Comment (LucaMele) [reply src/components/m-form-group/index.scss]: no nested selectors, always use BEM

File: src/components/a-checkbox/_preview.html [hunk 5]
```diff
@@ -1,2 +1,32 @@
...
-<label for="test">hello</label>
+<axa-checkbox
+  input-id="checkbox"
+  value="a"
+  name="test-name2"
+>
+  hello
+</axa-checkbox>
+<axa-checkbox
+  input-id="checkbox1"
```
Comment (LucaMele) [inline src/components/a-checkbox/_preview.html]: Please make id optional and as default use the UUID generator. If a consumer copies this code, he will copy the id name and therefore for more than one element we will have the same ID and therefore the HTML is invalid.

so please remove input-id in the preview.html s and add something like this in the template:

```
import html from 'nanohtml';
import BaseComponent from '../../abstarct/base-component';

export default function ({
  inputId = BaseComponent.uuid(),
  value,
  name,
  checked = false,
  disabled = false,
}, childrenFragment) {
  return html`<label class="a-radio__wrapper">
    <input
      id="${inputId}"
      class="a-radio__input"
      type="radio"
      name="${name}"
      checked="${checked}"
      value="${value}"
      disabled="${disabled}">
    <span class="a-radio__icon"></span>${childrenFragment}
  </label>`;
}

```

In this way, only in EXEPTIONAL cases id will be set from outside and we have just to trust that the user knows HTML

File: src/components/a-checkbox/_template.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+export default function ({
+  inputId,
+  value,
+  name,
+  checked = false,
+  disabled = false,
+}, childrenFragment) {
+  return html`<label class="a-checkbox__wrapper">
+    <input
+      id="${inputId}"
```
Comment (LucaMele) [inline src/components/a-checkbox/_template.js]: dtto ID

File: src/components/a-choice/_preview.html [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+</axa-choice>
+<axa-choice
+  input-id="choice3"
+  name="test-choice"
+  disabled
+>
+  Miss
+</axa-choice>
+<axa-choice
+  input-id="choice4"
```
Comment (LucaMele) [inline src/components/a-choice/_preview.html]: dtto ID

File: src/components/a-choice/_template.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+export default function ({
+  inputId,
+  value,
+  name,
+  checked = false,
+  disabled = false,
+}, childrenFragment) {
+  return html`<label class="a-choice__wrapper">
+    <input
+      id=${inputId}
```
Comment (LucaMele) [inline src/components/a-choice/_template.js]: dtto ID

File: src/components/a-form-input/_template.js (hunk 3 above)
Comment (LucaMele) [reply src/components/a-form-input/_template.js]: agree we have already a radio component

File: src/components/a-radio/_template.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+export default function ({
+  inputId,
+  value,
+  name,
+  checked = false,
+  disabled = false,
+}, childrenFragment) {
+  return html`<label class="a-radio__wrapper">
+    <input
+      id="${inputId}"
```
Comment (LucaMele) [inline src/components/a-radio/_template.js]: dtto ID

File: src/components/a-checkbox/_preview.html (hunk 5 above)
Comment (LucaMele) [reply src/components/a-checkbox/_preview.html]: but it is still `id="${inputId}"` here on the <input> tag and that is my problem

Comment (LucaMele) [reply src/components/a-checkbox/_preview.html]: People will copy paste html for the WC so in my opinion better to have a defaul UUID ID and only if needed set the id outside

Comment (LucaMele) [reply src/components/a-checkbox/_preview.html]: having same id on input field is one of the most common errors done in HTML programming due to copy pasta and to "but it works so what".

Comment (AndyOGo) [reply src/components/a-checkbox/_preview.html]: Sorry, my bad I missread the code section. Your are absolutely right.

**Comment by TheDadi** [comment]
@AndyOGo @LucaMele Are we ready to merge?

**Comment by LucaMele** [comment]
Yes :)

**Comment by LucaMele** [comment]
@TheDadi merge merge merge ...

**Pull request merged**